docs/CHANGELOG.md

```diff
@@ -1,5 +1,37 @@
 # Changelog
 
+## v0.17.0
+**NOTE**: this will be the final release of tuf that supports Python 2.7.
+This is because Python 2.7 was marked [end-of-life](
+https://www.python.org/dev/peps/pep-0373/) in January of 2020, and
+since then several of tuf's direct and transient dependencies have stopped
+supporting Python 2.7.
```

Comment:
Should we add max version constraints to requirements.txt and setup.py? Without constraints people won't be able to install TUF 0.17.0 dependencies on Python 2.7 as soon as any of the dependencies releases a newer version without Python 2.7 support.

Reply:
That's a good point. This might be a good justification for a release branch too, as dependencies drop Python 2.7 support we can add the max version constraint in the release branch for people who continue to use Python 2.7.

Reply:
I think that would mean that we would have to list our transitive dependencies in requirements.txt

Reply:
> I think that would mean that we would have to list our transitive dependencies in requirements.txt

Agreed. `requirements.txt` and `requirements-pinned` would list the same dependencies but with different constraint markers. 

```
# in requirements.txt

dependency1 < X.Y.Z    # no py2 above X.Y.Z
dependency2            # hasn't dropped py2 yet
```
```
# in requirements-pinned.txt 
dependency1 < X.Y.Z      # ignored by dependabot 
dependency2 == Y.V.W.    # bump with dependabot until py2, then constrain here and in requirements.txt
```

This definitely calls for a py2 branch

Reply:
What are the advantages of having a release branch compared to what we do now?

Reply:
Once we make the release we don't want to add any new features for Python 2.7, only fixes. If we do need to add fixes for Python2.7 we can create a branch at the point we tagged the release and add fixes there which are likely to be changes that don't exist in the develop branch, or changes implemented differently to similar changes in the develop branch.

We would usually release fixes and features together in a new release, the release branch enables us to release fixes without features and without having to hold features from landing in the develop branch.

Reply:
I guess these decisions are mostly up to whoever cares about and maintains the 0.17 (or python2) branch.

Or is there a suggestion here that we should do something now, before this release?

Reply:
> I guess these decisions are mostly up to whoever cares about and maintains the 0.17 (or python2) branch.
> 
> Or is there a suggestion here that we should do something now, before this release?

I think @lukpueh's concern is that if we don't do something _before_ the release, then a `pip install tuf` with python 2 can easily be broken when one of our dependencies (direct or transitive) updates and drops Python2 support.

I think the only way to resolve this is to list version constraints in our setup.py, because the install_requires lines in the setup.py are what pip uses to install dependencies [[1]](https://packaging.python.org/discussions/install-requires-vs-requirements/#id5).

Reply:
That was indeed my concern, @joshuagl. And, yes, we'd need to do in `setup.py` what I sketched out for `requirements.txt` above. We can even get rid of `requirements.txt`.

I'd still keep `requirements-pinned.txt` though, to track any future py2 drops in direct and transitive dependencies, using Dependabot and automatic builds, so that we can little by little constrain all direct and transitive dependencies in `setup.py` in future patch releases on a py2 release branch.

Alternatively, we can add all direct and transitive dependencies to `setup.py` and constrain them right away, and really make this the last Py2 release period. This would mean no more py2 work for us, which seems reasonable, but also doesn't allow py2 tuf users install any newer dependency updates that might still support py2.